libc/startup/baremetal/aie2/ldfile

```diff
@@ -0,0 +1,37 @@
+/*
+ * This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+ * See https://llvm.org/LICENSE.txt for license information.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ * (c) Copyright 2024 Advanced Micro Devices, Inc. or its affiliates
+ */
+
+MEMORY
+{
+   program (RX) : ORIGIN = 0, LENGTH = 0x0020000
+   data (!RX) : ORIGIN = 0x20000, LENGTH = 0x0020000
+}
+ENTRY(_main_init)
+SECTIONS
+{
+  . = 0x0;
+  .text : { 
+    *crt0.o(.text)
+    *(.text)
+  } > program
+
+  .stack : {
+    . = 0x20000;
+    _sp_start_value_DM_stack = .;
+    . += 0x2000;
+  } > data
+  .rodata : {
+    *(.rodata.*)
+  } > data
+  .data : {
+    *(.data)
```

Comment:
I think this should be *(.data`*`) ?

Reply:
I am not sure if we require `*` for .bss as well. Safe to keep it?

Reply:
I'll add wildcards everywhere. I think we have different sections for different alignments.